bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -272,19 +233,10 @@ def get_inputfiles(self):
 
         """
         firstarg = self.args[0]
-        if isinstance(firstarg, ROOT.TChain):
+        if isinstance(firstarg, ROOT.TTree):
             # Extract file names from a given TChain
-            chain = firstarg
-            return [chainElem.GetTitle()
-                    for chainElem in chain.GetListOfFiles()]
-        elif isinstance(firstarg, ROOT.TTree):
-            # Retrieve the associated file
-            treefile = firstarg.GetCurrentFile()
-            if not treefile:
-                # The tree has no associated input file
-                return None
-            else:
-                return [treefile.GetName()]
+            filenames = ROOT.Internal.TreeUtils.GetFileNamesFromTree(firstarg)
```

Comment:
So this also works to get all the filenames of a TChain right?

Reply:
correct